etcdserver/api/v3rpc/key.go

```diff
@@ -134,6 +134,14 @@ func (s *kvServer) Compact(ctx context.Context, r *pb.CompactionRequest) (*pb.Co
 	return resp, nil
 }
 
+func (s *kvServer) Hash(ctx context.Context, r *pb.HashRequest) (*pb.HashResponse, error) {
+	resp, err := s.kv.Hash(ctx, r)
+	if err != nil {
+		return nil, togRPCError(err)
```

Comment:
`togRPCError` only handles known types, check 
https://github.com/coreos/etcd/blob/master/etcdserver/api/v3rpc/key.go#L279-L291. Other types will return as internal error type. I am not sure what you want to achieve here. Return as internal error type is probably just fine.